visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts

```diff
@@ -41,12 +41,29 @@ export class CodeMapLabelService implements CameraChangeSubscriber {
 		ThreeOrbitControlsService.subscribe(this.$rootScope, this)
 	}
 
-	//labels need to be scaled according to map or it will clip + looks bad
-	addLabel(node: Node, options: { showNodeName: boolean; showNodeMetric: boolean }, highestNodeInSet: number) {
-		const state = this.storeService.getState()
+	// Labels need to be scaled according to map or it will clip + looks bad
+	addLabel(node: Node, highestNodeInSet: number, enforceLabel = false) {
+		const { appSettings, dynamicSettings, treeMap } = this.storeService.getState()
 
-		const { scaling, layoutAlgorithm } = state.appSettings
-		const { margin, heightMetric } = state.dynamicSettings
+		const { scaling, layoutAlgorithm, showMetricLabelNodeName, showMetricLabelNameValue } = appSettings
+		const { margin, heightMetric } = dynamicSettings
+
+		let labelText = ""
+
+		if (showMetricLabelNodeName || (enforceLabel && !showMetricLabelNameValue)) {
```

Comment:
Can we please rename the variables?

```suggestion
		if (showNodeName || (enforceLabel && !showMetric)) {
```

Reply:
I agree that new names would be better. This PR is already quite big though. I would rather do such change later.